### PR TITLE
Support for proxies

### PR DESCRIPTION
**Pull request by srikary12**
### Description
This pr adds support for Proxy in CLI

### Linked issues
closes <https://github.com/jozu-ai/kitops/issues/186>

### CODE REVIEW

**Comment by srikary12** [comment]
I'll check and update.

File: pkg/lib/constants/consts.go [hunk 1]
```diff
@@ -157,3 +157,18 @@ func TagIndexPathForRepo(storageBase, repo string) string {
...
+	} else if ProxyEnvVar = os.Getenv("https_proxy"); ProxyEnvVar != "" {
+		return "https_proxy"
+	} else if ProxyEnvVar = os.Getenv("HTTP_PROXY"); ProxyEnvVar != "" {
+		return "HTTP_PROXY"
+	} else if ProxyEnvVar = os.Getenv("http_proxy"); ProxyEnvVar != "" {
+		return "https_proxy"
+	}
+
+	return ProxyEnvVar
+}
```
[the remark this answers is not in the document]
Comment (srikary12) [reply pkg/lib/constants/consts.go:174]: I'll be purging this code.

File: pkg/cmd/options/network.go [hunk 2]
```diff
@@ -72,5 +74,6 @@ func DefaultNetworkOptions(configHome string) *NetworkOptions {
 		PlainHTTP:       false,
 		TLSVerify:       true,
 		CredentialsPath: constants.CredentialsPath(configHome),
+		Proxy:           constants.ProxyEnvVar(),
```
[the remark this answers is not in the document]
Comment (srikary12) [reply pkg/cmd/options/network.go:77]: I don't think we need to this line as proxy from env is set by default. We are overriding other proxies when we set proxy via the flag, so shouldn't be an issue if we purge this line.

**Comment by srikary12** [comment]
@amisevsk Please check, fixed the linting and purged the unnecessary code.

**Review by amisevsk — APPROVED** [review]
LGTM, thanks!

**Pull request merged**